python/ray/data/aggregate.py

```diff
@@ -71,17 +79,168 @@ def __init__(self):
         )
 
 
+def _null_init(init: Callable[[KeyType], AggType], k: KeyType):
+    """
+    Initialize accumulation while handling nulls.
+
+    This adds on a has_data field that the accumulator uses to track whether an
+    aggregation is empty.
+
+    Args:
+        init: The core init function.
+        k: Groupby key.
+
+    Returns:
+        Initialized accumulation that can be fed into the aggregation's
+        accumulate function.
+    """
+    a = init(k)
+    if not isinstance(a, list):
+        a = [a]
+    return a + [0]
```

Comment:
Could we add a type alias for the return type here? Like `NullableValue = Tuple[T, bool]` and use that throughout?

Reply:
There doesn't appear to be a straightforward such type to use, since it's actually closer to `MaybeAgg = AggType + Tuple[int]` (if `AggType` were a tuple), which I don't think that we can easily represent as a type. 🤔